### PR TITLE
[esxi-exporter] Deployment per Collector

### PR DESCRIPTION
**Pull request by christopherhans**
@richardtief prepared my deployment file for the esxi-exporter per collector. 
Could you please have a look, if this is all I need?

We can also talk about it in person.

### CODE REVIEW

File: prometheus-exporters/esxi-exporter/templates/service.yaml [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+---
 apiVersion: v1
 kind: Service
 metadata:
-  name: esxi-exporter
+  name: esxi-exporter-{{ required "$collector is missing" $collector }}
   namespace: infra-monitoring
   annotations:
     prometheus.io/scrape: "true"
     prometheus.io/port: {{ .Values.listen_port | quote }}
```
Comment (richardtief) [inline prometheus-exporters/esxi-exporter/templates/service.yaml:10]: at any place you had to put a`$.Values` to set it globally.

File: prometheus-exporters/esxi-exporter/templates/deployment.yaml [hunk 2]
```diff
@@ -6,31 +8,34 @@ metadata:
...
   template:
     metadata:
       labels:
-        app: esxi-exporter
+        app: esxi-exporter-{{ required "$collector is missing" $collector }}
         type: exporter
-      annotations:
-        prometheus.io/scrape: "true"
-        prometheus.io/port: {{ required ".Values.listen_port missing" .Values.listen_port | quote }}
-        prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
```
Comment (viennaa) [inline prometheus-exporters/esxi-exporter/templates/deployment.yaml:17]: How are you scraping now? Don't see any SD configuration change in this PR

Comment (richardtief) [reply prometheus-exporters/esxi-exporter/templates/deployment.yaml:17]: We decided to use kubernetes SD. So no default config is necessary.

Comment (viennaa) [reply prometheus-exporters/esxi-exporter/templates/deployment.yaml:17]: kubernetes or prometheus?

Comment (richardtief) [reply prometheus-exporters/esxi-exporter/templates/deployment.yaml:17]: we plan to use the kubernetes sd for prometheus like the vrops exporter to set variable scrape times.

File: system/infra-monitoring/templates/_prometheus-infra-collector.yaml.tpl [hunk 3]
```diff
@@ -617,23 +617,69 @@
...
+          names:
+            - infra-monitoring
   file_sd_configs:
       - files :
         - /etc/prometheus/configmaps/atlas-netbox-sd/netbox.json
   metrics_path: /
   relabel_configs:
+    - source_labels: [__meta_kubernetes_service_name]
+      action: keep
+      regex: esxi-exporter-Config*
```
Comment (richardtief) [inline system/infra-monitoring/templates/_prometheus-infra-collector.yaml.tpl:637]: A dot is missing here. 
`regex: esxi-exporter-Config.*`

File: system/infra-monitoring/templates/_prometheus-infra-collector.yaml.tpl [hunk 4]
```diff
@@ -617,23 +617,69 @@
...
+          names:
+            - infra-monitoring
+  file_sd_configs:
+      - files :
+        - /etc/prometheus/configmaps/atlas-netbox-sd/netbox.json
+  metrics_path: /
+  relabel_configs:
+    - source_labels: [__meta_kubernetes_service_name]
+      action: keep
+      regex: esxi-exporter-Critical*
```
Comment (richardtief) [inline system/infra-monitoring/templates/_prometheus-infra-collector.yaml.tpl:670]: and here.

**Review by richardtief — COMMENTED** [review]
I guess you have to run a `helm dependency update` to resolve outdated dependencies. Look at the checks.

**Comment by christopherhans** [comment]
oops

**Comment by christopherhans** [comment]
Any Git tips to solve the dependency errors without adding ~40 commits from the master branch into this PR?

**Review by viennaa — APPROVED** [review]

**Pull request merged**